[WIP] Activate doctest (#3503)
qgis/QGIS-Documentation: source/docs/pyqgis_developer_cookbook/crs.rst

* specify CRS by its ID

.. testsetup::

   from qgis.core import QgsCoordinateReferenceSystem

.. testcode::

   # PostGIS SRID 4326 is allocated for WGS84

>    crs = QgsCoordinateReferenceSystem("EPSG:4326")  # , QgsCoordinateReferenceSystem.PostgisCrsId)

Reviewer: ```suggestion
   crs = QgsCoordinateReferenceSystem("EPSG:4326") 
```

QGIS uses three different IDs for every reference system:

  * :const:`PostgisCrsId` --- IDs used within PostGIS databases.

>   * :const:`InternalCrsId` --- IDs internally used in QGIS database.

Reviewer: ```suggestion
  * :const:`InternalCrsId` --- IDs used in QGIS databases.
```

  * :const:`EpsgCrsId` --- IDs assigned by the EPSG organization

> If not specified otherwise in second parameter, PostGIS SRID is used by default.

Reviewer: ```suggestion
If not specified otherwise with the second parameter, PostGIS SRID is used by default.
```

* specify CRS by its ID

.. testsetup::

   from qgis.core import QgsCoordinateReferenceSystem

.. testcode::

   # PostGIS SRID 4326 is allocated for WGS84
   crs = QgsCoordinateReferenceSystem(4326, QgsCoordinateReferenceSystem.PostgisCrsId)

>    assert crs.isValid()

Reviewer 2: I know this is to run the test, but either we try to hide the last line (but test it anyway), or we need to explain what it does.

Reviewer 3: agreed, thank to have noticed it. Don't now the best solution: if we add a comment I'm afraid many users won't read it. @arnaud-morvan are there some workarounds? Like running all the `assert ... ` code during the building without having them explicit in the code snippets?

Author: We could put it in testcleanup, but it is a good practice to check if QgsCoordinateReferenceSystem is valid after creation. IMHO, last line should appear in documentation.

Reviewer 2: We can put something like this later:

The ``assert crs.isValid()`` line is used to confirm the instance validity, which is
considered a good practice after its creation.

Reviewer 3: +1 in fact `assert crs.isValid()` is not giving any results in the console.
